src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java

```diff
@@ -396,8 +395,13 @@ public String getTestJavaVersion() throws IOException {
             LOGGER.info("testJdkHome unset, using java available from the PATH");
             javaCmdAbsolutePath = "java";
         }
-        final Process process = new ProcessBuilder().command(javaCmdAbsolutePath, "-XshowSettings:properties -version").redirectErrorStream(true).start();
-        final String javaVersionOutput = IOUtils.toString(process.getInputStream());
+        final Process process = new ProcessBuilder().command(javaCmdAbsolutePath, "-XshowSettings:properties", "-version").redirectErrorStream(true).start();
+        try {
+            process.waitFor();
```

Comment:
this introduces a potential bug.

depending on the OS's buffer size and bufferingimplementaion the process can  be blocked from writing after the buffer is full, and as nothing is reading this output - the process will not complete and never finish.

in other words you should never `waitFor` a process without starting a draining (reading) thread first.

Additionally - and this is OT - if we are going to wait (possibly later) I would expect us to check the exit code and then fail if the command did not run correctly (due to a unsupported `X` operation on some non standard JVM)?

Reply:
Yeah, I am familiar with this problem and was just being lazy because the output is really small here (one line) and in practice the size of the pipe in Linux is 64KB. :-) If you feel strongly about it I'll add a gobbler thread.

Reply:
I don't think we need the `waitFor` here - it is mostly redundant as it - or it can be called after we have consumed the processes output if checking the exit code or ensuring the process is not dangling (should it close streams but not terminal) is required.